### PR TITLE
fix: decrease ctrlr_loss_timeout_sec for base bdev 

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#9874

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

File: pkg/spdk/engine.go [hunk 1]
```diff
@@ -1954,8 +1963,10 @@ func (e *Engine) BackupRestoreFinish(spdkClient *spdkclient.Client) error {
 			return err
 		}
 		e.log.Infof("Attaching replica %s with address %s before finishing restoration", replicaName, replicaAddress)
-		_, err = spdkClient.BdevNvmeAttachController(replicaName, helpertypes.GetNQN(replicaName), replicaIP, replicaPort, spdktypes.NvmeTransportTypeTCP, spdktypes.NvmeAddressFamilyIPv4,
-			helpertypes.DefaultCtrlrLossTimeoutSec, helpertypes.DefaultReconnectDelaySec, helpertypes.DefaultFastIOFailTimeoutSec, helpertypes.DefaultMultipath)
+		_, err = spdkClient.BdevNvmeAttachController(replicaName, helpertypes.GetNQN(replicaName), replicaIP, replicaPort,
+			spdktypes.NvmeTransportTypeTCP, spdktypes.NvmeAddressFamilyIPv4,
+			helpertypes.DefaultReplicaCtrlrLossTimeoutSec, helpertypes.DefaultReplicaReconnectDelaySec, helpertypes.DefaultReplicaFastIOFailTimeoutSec,
+			helpertypes.DefaultMultipath)
```
[the remark this answers is not in the document]
Comment (derekbit) [reply pkg/spdk/engine.go]: Updated.

File: pkg/spdk/engine.go [hunk 2]
```diff
@@ -1078,7 +1086,8 @@
 	}
 
 	// Add rebuilding replica head bdev to the base bdev list of the RAID bdev
-	dstHeadLvolBdevName, err := connectNVMfBdev(spdkClient, dstReplicaName, dstHeadLvolAddress)
+	dstHeadLvolBdevName, err := connectNVMfBdev(spdkClient, dstReplicaName, dstHeadLvolAddress,
+		helpertypes.DefaultReplicaCtrlrLossTimeoutSec, helpertypes.DefaultReplicaFastIOFailTimeoutSec)
```
[the remark this answers is not in the document]
Comment (derekbit) [reply pkg/spdk/engine.go]: updated.

**Comment by derekbit** [comment]
Ready for review cc @shuo-wu @c3y1huang @DamiaSan @innobead

File: pkg/spdk/types.go [hunk 3]
```diff
@@ -31,6 +32,16 @@ const (
 	retryInterval = 1 * time.Second
 )
 
+const (
+	// Timeouts for RAID base bdev (replica)
+	replicaCtrlrLossTimeoutSec  = 15
```
Comment (c3y1huang) [inline pkg/spdk/types.go:37]: Could we add a comment explaining how this value is determined?

Comment (derekbit) [reply pkg/spdk/types.go:49]: Added. Thank you.

**Review by c3y1huang — APPROVED** [review]

**Pull request merged**